include/pybind11/detail/class.h

```diff
@@ -605,7 +604,6 @@ extern "C" inline int pybind11_getbuffer(PyObject *obj, Py_buffer *view, int fla
     // Fill in all the information, and then downgrade as requested by the caller, or raise an
     // error if that's not possible.
     view->obj = obj;
-    view->internal = info;
```

Comment:
This looks good, but feels a little inconsistent, why move setting `view->internal` to the end while setting `view->buf` here from the same `info` object? I would keep them both together, and it would be good to explain why it is moved to the end if that is preferred.

I think in the case of error `info` is deleted and also `info->ptr`.